server/src/main/java/keywhiz/service/resources/automation/v2/SecretResource.java

```diff
@@ -815,6 +817,17 @@ public Iterable<String> modifySecretGroups(@Auth AutomationClient automationClie
         .collect(toSet());
   }
 
+  @Timed
+  @ExceptionMetered
+  @GET
+  @Path("/deleted/{name}")
+  @LogArguments
+  public List<SecretSeries> getDeletedSecretSeriesByName(
+    @Auth AutomationClient automationClient,
+    @PathParam("name") String name) {
+    return secretDAO.getSecretsWithDeletedName(name);
+  }
```

Comment:
Easy peasy. This functionality was already available but only exposed to the CLI, so exposing it via a REST endpoint was trivial.